models/block.go

```diff
@@ -73,11 +72,13 @@ func (b *Block) Hash() (gethCommon.Hash, error) {
 func decodeBlockEvent(event cadence.Event) (*Block, error) {
 	payload, err := events.DecodeBlockEventPayload(event)
 	if err != nil {
-		if block, err := decodeLegacyBlockEvent(event); err == nil {
-			return block, nil
-		}
+		return nil, err
+	}
 
-		return nil, fmt.Errorf("failed to cadence decode block [%s]: %w", event.String(), err)
+	var fixedHash *string
+	if payload.PrevRandao == gethCommon.HexToHash("0x0") {
+		hash := payload.Hash.String()
+		fixedHash = &hash
```

Comment:
not sure why we populate it using this heuristic?

Reply:
This change is not really related to the objective of the PR, but it is required due to upgrading the `flow-go` versioned.
This heuristic allows us to decode block events prior to the addition of the `PrevRandao` field. The decoding of legacy block events was moved in flow-go (https://github.com/onflow/flow-go/blob/master/fvm/evm/events/events.go#L188-L190), so we no longer have a way to distinguish them, hence the usage of this heuristic. I'll add a comment to describe this though, as it is for sure not clear why it's needed :sweat_smile: